src/modules/esl-media/core/esl-media.ts

```diff
@@ -246,9 +246,13 @@ export class ESLMedia extends ESLBaseElement {
     return this._provider && this._provider.safeStop();
   }
 
-  /** Toggle play/pause state of the media */
-  public toggle(): Promise<void> | null {
-    return this._provider && this._provider.safeToggle();
+  /**
+   * Executes toggle action:
+   * If the player is PAUSED then it starts playing otherwise it pause playing
+   */
+  public toggle(allowActivate: boolean = false): Promise<void> | null {
+    const shouldActivate = [PlayerStates.PAUSED, PlayerStates.UNSTARTED, PlayerStates.VIDEO_CUED].includes(this.state);
```

Comment:
There is no sense in planning `pause` action for a video that was not load the provider
```suggestion
    const shouldActivate = [PlayerStates.PAUSED, PlayerStates.UNSTARTED, PlayerStates.VIDEO_CUED, PlayerStates.UNINITIALIZED].includes(this.state);
```